homework/src/App.js

```diff
@@ -5,77 +5,72 @@
 class App {
   constructor(url) {
     this.initialize(url);
+    this.root = document.getElementById('root');
+    this.header = Util.createAndAppend('header', this.root);
+    this.label = Util.createAndAppend('label', this.header, { text: 'HYF Repositories' });
+    this.select = Util.createAndAppend('select', this.header);
+    this.article = Util.createAndAppend('article', this.root);
+    this.leftDiv = Util.createAndAppend('div', this.article, { id: 'leftDiv' });
+    this.rightDiv = Util.createAndAppend('div', this.article, { id: 'rightDiv' });
   }
-
-  /**
-   * Initialization
-   * @param {string} url The GitHub URL for obtaining the organization's repositories.
-   */
   async initialize(url) {
-    // Add code here to initialize your app
-    // 1. Create the fixed HTML elements of your page
-    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
-
-    const root = document.getElementById('root');
-    // ...
-
     try {
-      // ...
       const repos = await Util.fetchJSON(url);
-      this.repos = repos.map(repo => new Repository(repo));
-      // ...
-    } catch (error) {
+      this.repos = repos.map(repo => new Repository(repo));  // turning each repo into a class repo
+      repos.sort((a, b) => a.name.localeCompare(b.name));
+      repos.forEach((repository, index) => {
+        Util.createAndAppend('option', this.select, {
+          text: repository.name,
+          value: index
+        });
+      });
```

Comment:
There is a mismatch between the sort order of the `option` children of the `select` element and the array of repositories that you store in `this.repos`. Therefore the `index` that you store in line 24 point to the wrong `Repository` object from the `this.repos` array. 

For instance, if I start your application I see '**alumni**' in the select box but the repo displayed is the '**AngularJS**' repo. (Did you not see this?)

Can try and fix this?